src/js/media-live-button.js

```diff
@@ -0,0 +1,89 @@
+import MediaChromeButton from './media-chrome-button.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import {
+  Window as window,
+  Document as document,
+} from './utils/server-safe-globals.js';
+import { MediaUIEvents, MediaUIAttributes } from './constants.js';
+import { verbs } from './labels/labels.js';
+
+const { MEDIA_TIME_IS_LIVE, MEDIA_PAUSED, MEDIA_STREAM_TYPE } = MediaUIAttributes;
+const { MEDIA_SEEK_TO_LIVE_REQUEST, MEDIA_PLAY_REQUEST } = MediaUIEvents;
+
+const indicatorSVG = '<svg viewBox="0 0 4 4"><circle cx="2" cy="2" r="2"></circle></svg>';
+
+const slotTemplate = document.createElement('template');
+slotTemplate.innerHTML = `
+  <style>
+
+  :host {
+    text-transform: uppercase;
+  }
+
+  slot[name=indicator] > *,
+  :host ::slotted([slot=indicator]) {
+    padding: var(--media-live-indicator-icon-padding, 0 5px 0 0);
+
+    /* Override styles for icon-only buttons */
+    min-width: auto;
+
+    /* svgs */
+    fill: var(--media-live-indicator-icon-color-off, rgb(118, 118, 118));
+    height: var(--media-live-indicator-icon-height, 5px);
+    width: var(--media-live-indicator-icon-width, 5px);
+    
+    /* font icons */
+    color: var(--media-live-indicator-icon-color-off, rgb(118, 118, 118));
+    font-size: 5px;
+    line-height: 5px;
+  }
+
+  :host([${MEDIA_TIME_IS_LIVE}]) slot[name=indicator] > *,
+  :host([${MEDIA_TIME_IS_LIVE}]) ::slotted([slot=indicator]) {
+    fill: var(--media-live-indicator-icon-color, rgb(255, 0, 0));
+    color: rgb(255, 0, 0);
+  }
+  
+  </style>
+
+  <slot name="indicator">${indicatorSVG}</slot>
+  <slot name="text">Live</slot>
+`;
+
+class MediaLiveButton extends MediaChromeButton {
+  static get observedAttributes() {
+    return [
+      ...super.observedAttributes,
+      MEDIA_PAUSED,
+      MEDIA_TIME_IS_LIVE
```

Comment:
suggestion(non-blocking): We should add `MEDIA_STREAM_TYPE` to the list of observed attributes to make styling cases easier.

Reply:
👍 done